gapic/src/main/com/google/gapid/models/Profile.java

```diff
@@ -412,6 +351,49 @@ public boolean hasChildren() {
     public List<PerfNode> getChildren() {
       return children;
     }
+
+    protected void addChild(PerfNode node) {
+      children.add(node);
+    }
+
+    protected PerfNode findNode(Path.Commands link, int level) {
+      int idx = Collections.binarySearch(children, null, (node, $) -> {
+        Path.Commands current = node.group.getLink();
+        int r = 0;
+        for (int i = 0; r == 0 && i < level; i++) {
+          r = Long.compare(current.getFrom(i), link.getFrom(i));
+          if (r == 0) {
+            r = Long.compare(current.getTo(i), link.getTo(i));
+          }
+        }
+        return r;
+      });
+      return idx < 0 ? null : children.get(idx);
+    }
+
+    protected PerfNode findNodeContaining(Path.Commands link) {
```

Comment:
From the name I'm not sure what differentiates this function from the findNode above. Can you explain the difference in a comment so people know which to use and what their difference is?

Reply:
Added comments.